### PR TITLE
Move serve functionality out of allennlp command.

### PR DESCRIPTION
**Pull request by schmmd**
The `allennlp` command currently requires dependencies that few of our users will use and that increase the time it takes to run `allennlp`. Furthermore, the serve command does not work from pip installs, which is the main way our users install AllenNLP.

This PR accomplishes this by moving functionality from `commands` to `allennlp/service/server_flask.py`.

Fixes https://github.com/allenai/allennlp/issues/1303

### CODE REVIEW

**Comment by schmmd** [comment]
I see this as a stepping stone to moving the demo into its own repository.  Although I think that's blocked on publishing a Docker image that include `allennlp` from a pip install.

**Comment by schmmd** [comment]
When this is merged I will need to update https://github.com/allenai/allennlp-internal/blob/master/scripts/deploy/deploy-staging-demo.sh and https://github.com/allenai/allennlp-internal/blob/master/scripts/deploy/deploy-production-demo.sh.

File: allennlp/service/predictors/predictor.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # pylint: disable=unused-import
 import warnings
 
-from allennlp.predictors.predictor import Predictor, DemoModel, DEFAULT_PREDICTORS
+from allennlp.predictors.predictor import Predictor, DEFAULT_PREDICTORS
```
Comment (DeNeutoy) [inline allennlp/service/predictors/predictor.py]: Leave the DemoModel import here (change it to it's new location) to preserve backwards compatibility

File: allennlp/service/predictors/__init__.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 """
 import warnings
 
-from allennlp.predictors.predictor import Predictor, DemoModel
+from allennlp.predictors.predictor import Predictor
```
Comment (DeNeutoy) [inline allennlp/service/predictors/__init__.py]: Same here

File: allennlp/service/predictors/predictor.py (hunk 1 above)
Comment (schmmd) [reply allennlp/service/predictors/predictor.py]: Ultimately I'd like to move the demo to a separate repository that depends on AllenNLP by a pip dependency.  How can I leave the import here but mark it as deprecated?

Comment (DeNeutoy) [reply allennlp/service/predictors/predictor.py]: copy the source here or something? 🤷‍♂️

File: allennlp/service/predictors/__init__.py (hunk 2 above)
Comment (schmmd) [reply allennlp/service/predictors/__init__.py]: Done

File: allennlp/service/predictors/predictor.py (hunk 1 above)
Comment (schmmd) [reply allennlp/service/predictors/predictor.py]: I added it to `__init__.py` but I don't understand how adding that import would help backwards compatibility here.  Can you explain/give an example?

**Pull request merged**